Level 1

KorolevaAnastasia opened the pull request:
dev:
развернут бэк, проверки функций выполнены
создание пользователя, изменение пользователя
создание фильма, удаление фильма, получение фильмов текущего пользователя
регистрация, авторизация

KorolevaAnastasia committed:
initial commit:
добавлены модель+контроллеры пользователя и фильмов, прописаны роуты

KorolevaAnastasia committed:
fix: изменение названия бд

KorolevaAnastasia committed:
fix: проверка данных при добавлении фильма

KorolevaAnastasia committed:
fix: проверка данных при добавлении фильма (передача id пользователя)

KorolevaAnastasia committed:
fix: проверка данных при получении списка фильмов (передача id пользователя)

KorolevaAnastasia committed:
fix: проверка удаления чужих карточек

KorolevaAnastasia committed:
fix: добавлены ссылки, проверки пройдены

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [ ] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [ ] `PATCH /users/me` — обновляет информацию о пользователе;
  **Запрос не работает.** 
![2023-05-23_04-50-45](https://github.com/KorolevaAnastasia/movies-explorer-api/assets/86304001/41ffddc4-df4f-4e09-9d6a-9bb08019845f)
  - [ ] `GET /movies` — все сохранённые пользователем фильмы;
  **Невозможно проверить, так как фильм не создается.**
  - [ ] `POST /movies` — создаёт фильм с переданными в теле данными;
  **Запрос не работает.**  
![2023-05-23_04-51-27](https://github.com/KorolevaAnastasia/movies-explorer-api/assets/86304001/489f9448-a4d4-4d2c-acde-f65329d1d58b)
  - [ ] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  **Невозможно проверить, так как фильм не создается.**
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [ ] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
**Невозможно проверить, так как фильм не создается.**
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [ ] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [ ] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [ ] сообщение об ошибке соответсвует ее типу;
	**Не обработана ошибка дубля email при редактировании пользователя.**
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [ ] <!--7.28--> Валидация реализована верно:
  - [ ] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации. 
  **1. Параметр `movieId` в запросе удаления фильма не валидируется.**
  **2. Поле `name` в запросе регистрации должно быть обязательным.**
  **3. Неверная валидация запроса редактирования пользователя.**
  **4. Все поля в запросе редактирования пользователя должны быть обязательными.**
  **5. Лишние ограничения длины для полей фильма в запросе создания фильма (нет в ТЗ).**
  - [ ] Данные валидируются перед добавлением в базу. 
  **Поле `name` на уровне схемы пользователя не должно иметь значения по умолчанию.**
- [ ] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
**Адрес общий для всех режимов.**
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [ ] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [ ] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  **1. Файл `.env` в репозитории.**
  **2. Файл `.env` добавляется в репозиторий.**
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики
- [x] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [x] <!--2.14--> Валидация описана в отдельном модуле.
- [ ] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [ ] файлы логов не добавляются в репозиторий.
  **Файлы логов добавляются в репозиторий.**
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [ ] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
**Модуль не используется.**
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [x] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
  **Сообщения прописаны вручную.**
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
**Модуль не используется.**
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
**Модуль не используется.**
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 40

KorolevaAnastasia committed:
fix:
отредактирован gitignore
тексты ошибок вынесены в константы
поправлена валидация
сохранение файлов логов вынесено в отдельную папку
поправлена ошибка в названии роутов movies

KorolevaAnastasia committed:
fix:
файлы логирования

KorolevaAnastasia committed:
fix:
файлы логирования - проверка

KorolevaAnastasia committed:
fix:
корректировка валидации для удаления фильма

KorolevaAnastasia commented:
-  поправлен gitignore (папка логов и .env не добавляются в репозиторий)
-  текст ошибок перенесен в отдельные константы
- исправлена ошибка в роуте movies
- изменение валидации данных для пользователя и фильмов (при добавлении в базу)

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [ ] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
**Есть ошибка.**
![2023-05-24_03-56-04](https://github.com/KorolevaAnastasia/movies-explorer-api/assets/86304001/2e300d56-0f79-41de-a832-44fd8ae01698)
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [ ] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [x] `GET /movies` — все сохранённые пользователем фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [ ] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  **Запрос не работает.**  
![2023-05-24_03-54-12](https://github.com/KorolevaAnastasia/movies-explorer-api/assets/86304001/80b1b4fb-70f6-4c39-b4a7-25315d4a59af)
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [ ] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
**Невозможно проверить, так как фильм не удаляется.**
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [x] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [x] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [x] сообщение об ошибке соответсвует ее типу;
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [ ] <!--7.28--> Валидация реализована верно:
  - [ ] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации. 
  **Неверная валидация поля `email` в запросе редактирования пользователя.**
  - [x] Данные валидируются перед добавлением в базу. 
- [x] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [x] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики
- [x] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [x] <!--2.14--> Валидация описана в отдельном модуле.
- [x] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [ ] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
**Модуль не используется.**
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [x] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
  **Присутствуют сообщения, прописанные вручную.**
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
**Модуль не используется.**
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
**Модуль не используется.**
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 61

## Доп. Комментарий:
**У фильма нет уникальных полей.**
![2023-05-24_04-00-02](https://github.com/KorolevaAnastasia/movies-explorer-api/assets/86304001/5e910f17-bd78-4d07-b0cd-2e931daef0c3)

KorolevaAnastasia committed:
fix:
выполнена npx eslint
DELETE /movies/movieId — удаляет сохранённый фильмы по _id;
Исправлена неверная валидация поля email в запросе редактирования пользователя.
Токен не сохраняется в куках
Убрана ошибка конфликта

KorolevaAnastasia commented:
-  выполнена  npx eslint
-  DELETE /movies/movieId — удаляет сохранённый фильмы по _id;
-  Исправлена неверная валидация поля email в запросе редактирования пользователя.
-  Токен не сохраняется в куках
-  Убрана ошибка конфликта

StarchenkovYaroslav commented:
# Резюме:
## Backend

### Критерии, влияющие на работоспособность
- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [x] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [x] `GET /movies` — все сохранённые пользователем фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [x] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [x] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
	- [x] сообщение об ошибке соответсвует ее типу;
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [x] <!--7.28--> Валидация реализована верно:
  - [x] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации. 
  - [x] Данные валидируются перед добавлением в базу. 
- [x] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [x] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики
- [x] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [x] <!--2.14--> Валидация описана в отдельном модуле.
- [x] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.14--> API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации
- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [ ] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
**Модуль не используется.**
- [x] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [x] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [x] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
**Модуль не используется.**
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
**Модуль не используется.**
- [x] <!--0.71--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 98
